netket/experimental/operator/_fermions_2nd.py

```diff
@@ -0,0 +1,769 @@
+        if not isinstance(other, FermionOperator2nd):
+            return NotImplementedError
+        if not self.hilbert == other.hilbert:
+            raise ValueError(
+                f"Can only multiply identical hilbert spaces (got A@B, A={self.hilbert}, B={other.hilbert})"
+            )
+        terms = []
+        weights = []
+        for t, w in self._operators.items():
+            for to, wo in other._operators.items():
+                terms.append(tuple(t) + tuple(to))
+                weights.append(w * wo)
+        if other._constant != 0.0:
+            for t, w in self._operators.items():
+                terms.append(tuple(t))
+                weights.append(w * other._constant)
+        if self._constant != 0:
+            for t, w in other._operators.items():
+                terms.append(tuple(t))
+                weights.append(w * self._constant)
+        constant = self._constant * other._constant
+        dtype = np.promote_types(self.dtype, other.dtype)
+        return FermionOperator2nd(
+            self.hilbert, terms, weights=weights, constant=constant, dtype=dtype
+        )
+
+    def _iop__matmul__(self, other):
+        if not isinstance(other, FermionOperator2nd):
+            return NotImplementedError
+        if not self.hilbert == other.hilbert:
+            raise ValueError(
+                f"Can only multiply identical hilbert spaces (got A@B, A={self.hilbert}, B={other.hilbert})"
+            )
+        if not np.can_cast(_dtype(other), self.dtype, casting="same_kind"):
+            raise ValueError(
+                f"Cannot multiply inplace operator with dtype {type(other)} "
+                f"to operator with dtype {self.dtype}"
+            )
+
+        terms = []
+        weights = []
+        for t, w in self._operators.items():
+            for to, wo in other._operators.items():
+                terms.append(tuple(t) + tuple(to))
+                weights.append(w * wo)
+        if other._constant != 0.0:
+            for t, w in self._operators.items():
+                terms.append(tuple(t))
+                weights.append(w * other._constant)
+        if self._constant != 0:
+            for t, w in other._operators.items():
+                terms.append(tuple(t))
+                weights.append(w * self._constant)
+        constant = self._constant * other._constant
+
+        self._operators = dict(zip(terms, weights))
+        self._constant = constant
+        return self
+
+    def _op_matmul(self, other):
+        if not isinstance(other, FermionOperator2nd):
+            return NotImplementedError
+        dtype = np.promote_types(self.dtype, other.dtype)
+        op = self.copy(dtype=dtype)
+        op @= other
+        return op
+
+    def __radd__(self, other):
+        return self + other
+
+    def __add__(self, other):
+        dtype = np.promote_types(self.dtype, _dtype(other))
+        op = self.copy(dtype=dtype)
+        op += other
+        return op
```

Comment:
we already know that we'll call this method so let's bypass python
```suggestion
        return op.__iadd__(other)
```